src/js/modules/rpc/errors.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+import {
+  DisableCoprocessorData,
+  EnableCoprocessor,
+  EnableCoprocessorMetadataItem,
+  EnableCoprocessorRequestData,
+} from "../domain/generatedRpc/generatedClasses";
+import { Handle } from "../domain/Handle";
+import { PolicyInjection } from "../public/Coprocessor";
+import { Logger } from "winston";
+
+export enum EnableResponseCodes {
+  success,
+  internalError,
+  invalidIngestionPolicy,
+  scriptIdAlreadyExists,
+  scriptContainsInvalidTopic,
+  scriptContainsNoTopics,
+  scriptContainsSyntaxError,
+}
+
+export enum DisableResponseCode {
+  success,
+  internalError,
+  scriptDoesNotExist,
+}
+
+const maxSizeTopicName = 249;
+const validateKafkaTopicName = (topic: string): boolean => {
+  const validAlphaNumeric = /^[0-9a-zA-Z]+$/;
+  if (topic.length === 0) {
+    return false;
+  } else if (topic == "." || topic == "..") {
+    return false;
+  } else if (topic.length > maxSizeTopicName) {
+    return false;
+  } else if (!validAlphaNumeric.test(topic)) {
+    return false;
+  }
+  return true;
+};
+
+type EnableCoprocResponse<A> = (handle: A) => EnableCoprocessorRequestData;
+type HandleEnableResponse = EnableCoprocResponse<Handle>;
+type SimpleEnableResponse = EnableCoprocResponse<EnableCoprocessor>;
+type DisableCoprocResponse = (id: bigint) => DisableCoprocessorData;
+
+const validateLoadScriptError = (
+  e: Error,
+  id: bigint,
+  script: Buffer,
+  logger: Logger
+): EnableCoprocessorRequestData => {
+  if (e instanceof SyntaxError) {
+    logger.info(`syntax error on wasm script: ${id}, ${e.message}`);
+    return createResponseScriptSyntaxError({ id, source_code: script });
+  } else {
+    logger.info(`error on wasm script: ${id}, ${e.message}`);
```

Comment:
And this looks like it should be `logger.error`

Reply:
done